### PR TITLE
Add TOC navigation to environment subfolder Issue#854

### PR DESCRIPTION
**Pull request by FiniteStateGit**
Fixes [ISSUE][DOCS] Table of contents not enabled for pages in Vegetation Gem https://github.com/o3de/o3de.org/issues/854

Created new __index.md for the Environment subfolder, assigned weights for TOC and cleaned up frontmatter of directory.

### CODE REVIEW

File: content/docs/user-guide/gems/reference/environment/_index.md [hunk 1]
```diff
@@ -0,0 +1,13 @@
+---
+linkTitle: Environment
+title: Environment Gems
+description: The Environment Gems provide the tools and framework to create natural-looking vegetation in Open 3D Engine (O3DE) projects.
+---
+
+The dynamic vegetation system uses vegetation components to customize dynamic vegetation coverage for levels of any size. To use the dynamic vegetation system, you must enable the [Vegetation Gem](./vegetation) for your project.
+
+The [Landscape Canvas Gem](./landscape-canvas.md) provides the **Landscape Canvas** editor; a node-based graph tool for authoring workflows to populate landscape with dynamic vegetation.
```
Comment (sptramer) [inline content/docs/user-guide/gems/reference/environment/_index.md:9]:
```suggestion
The [Landscape Canvas Gem](./landscape-canvas) provides the **Landscape Canvas** editor, a node-based graph tool for authoring workflows to populate landscape with dynamic vegetation.
```

File: content/docs/user-guide/gems/reference/environment/_index.md [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+
+The [Vegetation Assets Gem](./vegetation-gem-assets.md) provides vegetation models, textures, and other assets and samples for use with the Vegetation Gem and **Landscape Canvas**.
```
Comment (sptramer) [inline content/docs/user-guide/gems/reference/environment/_index.md:11]:
```suggestion
The [Vegetation Assets Gem](./vegetation-gem-assets) provides vegetation models, textures, and other assets and samples for use with the Vegetation Gem and **Landscape Canvas**.
```

File: content/docs/user-guide/gems/reference/environment/_index.md [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+
+The [Surface Data Gem](./surface-data.md) enables surfaces such as terrain or meshes to emit signals or tags, that communicate its surface type. These signals and tags have various uses, such as creating inclusion and exclusion areas for vegetation growth.
```
Comment (sptramer) [inline content/docs/user-guide/gems/reference/environment/_index.md:13]:
```suggestion
The [Surface Data Gem](./surface-data) enables surfaces such as terrain or meshes to emit signals or tags, that communicate its surface type. These signals and tags have various uses, such as creating inclusion and exclusion areas for vegetation growth.
```

**Review by sptramer — CHANGES_REQUESTED** [review]
Some minor copy changes requested. Thanks for your contribution!

**Review by sptramer — APPROVED** [review]

**Review by micronAMZN — APPROVED** [review]

**Pull request merged**